ldm/invoke/app/services/graph.py

```diff
@@ -593,10 +597,18 @@ def complete(self, node_id: str, output: InvocationOutputsUnion):
         if all([n in self.executed for n in prepared_nodes]):
             self.executed.add(source_node)
             self.executed_history.append(source_node)
+    
+    def error(self, node_id: str, error: str):
```

Comment:
perhaps `mark_errored` is a better function name? i'd expect a function called `error` to return an error rather than changing state